gui/librarydialog.cpp

```diff
@@ -34,9 +34,9 @@
 class FunctionListItem : public QListWidgetItem {
 public:
     FunctionListItem(QListWidget *view,
-                     CppcheckLibraryData::Function *function,
+                     CppcheckLibraryData::Function *paramFunction,
```

Comment:
There are quite a few warnings of the type

`warning: constructor parameter '[variable name]' shadows the field '[variable name]' of '[class name]'[-Wshadow-field-in-constructor,Semantic Issue]`

in the code. I find the practice less than ideal and I have only seen it done in this project. Renaming the parameters to a function so that they are NOT the same as class variables seems best course.

Reply:
I can understand if you have a different opinion. please just use `-Wno-shadow-field-in-constructor`.